functionnal_tests/cypress/integration/app_folder/app_folder_spec.js

```diff
@@ -1,34 +1,83 @@
+import { SELECTORS as s } from '../../support/generic_selector_commands.js'
 
 describe('App Folder Advanced', function () {
-  beforeEach(function () {
+  let workspaceId
+  let folder1 = {label: 'first Folder'}
+  let folder2 = {label: 'second Folder'}
+
+  before(function () {
     cy.resetDB()
     cy.setupBaseDB()
     cy.loginAs('administrators')
     cy.fixture('baseWorkspace').as('workspace').then(workspace => {
+      workspaceId = workspace.workspace_id
+
       cy.visit(`/ui/workspaces/${workspace.workspace_id}/contents`)
+
+      cy.createFolder(folder1.label, workspaceId).then(f => (folder1 = f))
+      cy.createFolder(folder2.label, workspaceId).then(f => (folder2 = f))
     })
   })
 
+  beforeEach(() => {
+    cy.loginAs('administrators')
+  })
+
+  afterEach(() => {
+    cy.cancelXHR()
+  })
+
   it('should open when editing a folder', () => {
-    create_folder(cy)
-    open_app_advanced_folder(cy)
+    cy.visitPage({pageName: PAGES.CONTENTS, params: {workspaceId: workspaceId}})
+
+    cy.getTag({selectorName: s.FOLDER_IN_LIST, params: {folderId: folder1.content_id}})
+      .find('.extandedaction.dropdown')
+      .click()
+
+    cy.getTag({selectorName: s.FOLDER_IN_LIST, params: {folderId: folder1.content_id}})
+      .find('.extandedaction.dropdown .fa.fa-pencil')
+      .click()
 
     cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
       .should('be.visible')
   })
 
   it('should closed itself when clicking on the close button', () => {
-    create_folder(cy)
-    open_app_advanced_folder(cy)
+    cy.visitPage({pageName: PAGES.EDIT_FOLDER, params: {workspaceId: workspaceId, folderId: folder1.content_id}})
 
-    cy.get('[data-cy=popinFixed__header__button__close]')
-      .should('be.visible')
+    cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
+      .find('.folderAdvanced__header__title')
+      .contains(folder1.label)
+
+    cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
+      .find('[data-cy="popinFixed__header__button__close"]')
       .click()
 
     cy.get('#appFeatureContainer').children().should('have.length', 0)
   })
+
+  it('should reopen and reload data when opening a different folder', () => {
+    cy.visitPage({pageName: PAGES.EDIT_FOLDER, params: {workspaceId: workspaceId, folderId: folder1.content_id}})
+
+    cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
+      .find('.folderAdvanced__header__title')
+      .contains(folder1.label)
+
+    cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
+      .find('[data-cy="popinFixed__header__button__close"]')
+      .click()
+
+    cy.getTag({selectorName: s.FOLDER_IN_LIST, params: {folderId: folder2.content_id}})
+      .find('.extandedaction.dropdown')
+      .click()
+
+    cy.getTag({selectorName: s.FOLDER_IN_LIST, params: {folderId: folder2.content_id}})
+      .find('.extandedaction.dropdown .fa.fa-pencil')
```

Comment:
I think we need a data-cy for this selector, this is a generic component and we can't base our test on an icon.

Reply:
data-cy added